jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java

```diff
@@ -188,6 +190,27 @@ public AbstractConnector(
             pool = _server.getBean(ByteBufferPool.class);
         _byteBufferPool = pool != null ? pool : new ArrayByteBufferPool();
 
+        addEventListener(new Container.Listener()
+        {
+            @Override
+            public void beanAdded(Container parent, Object bean)
+            {
+                if (bean instanceof HttpChannel.Listener)
+                    // TODO evaluate which is faster?
+                    _httpChannelListeners = new HttpChannelListeners(getBeans(HttpChannel.Listener.class));
```

Comment:
The intention of HttpChannelListeners is for it to be as fast as possible to invoke - so that means there can be no add semantics as I want everything final within the class.  So makeing a new instance on each add is OK.